src/components/redux/reducer.js

```diff
@@ -0,0 +1,33 @@
+import {ADD_TODO, UPDATE_TODO, DELETE_TODO} from './action';
+import { todos } from './stats';
+
+export let reducer = (state = todos, action) => {
+    let newTodos;
+    switch (action.type) {
+        case ADD_TODO:
+            newTodos= [...state];
+            newTodos.push(action.payload);
+            return newTodos;
+        case DELETE_TODO:
+            newTodos= [...state];
+            newTodos= newTodos.filter(todo => todo.id != action.payload)
+            return newTodos;
+        case UPDATE_TODO:
+            newTodos= [...state];
+            let index= -1;
+            for (let i = 0; i < newTodos.length; i++) {
+                index ++;
+                if(newTodos[i].id == action.payload.id) {
+                    break;
+                }
+            }
+            if(index != -1) {
+                newTodos[index]= action.payload;
+                return newTodos;    
+
+            }
```

Comment:
It's a bad practice to leave a switch case without a break or return and depend on the next switch case to return a value